pytorch_lightning/trainer/training_loop.py

```diff
@@ -426,7 +426,9 @@ def run_training_epoch(self):
                 # logs user requested information to logger
                 self.log_metrics(batch_step_metrics, grad_norm_dic)
 
-            self.global_step += 1
+            # progress global step according to grads progress
+            if (self.batch_idx + 1) % self.accumulate_grad_batches == 0:
```

Comment:
we shall clarify what is the meaning of these two variables... 
could you pls ass comment to the trainer init starting with #: so it is generated also in the documentation...
 Thx